### PR TITLE
Fix error logging in jobs

### PR DESCRIPTION
**Pull request by skofman1**
@maartenba @xavierdecoster

### CODE REVIEW

File: src/Stats.CollectAzureCdnLogs/Job.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ public override bool Init(IDictionary<string, string> jobArgsDictionary)
             }
             catch (Exception ex)
             {
-                _logger.LogCritical("Job failed to initialize!", ex);
+                _logger.LogCritical("Job failed to initialize {Exception}!", ex);
```
Comment (scottbommarito) [inline src/Stats.CollectAzureCdnLogs/Job.cs]: `Job failed to initialize! {Exception}` is probably what you want here.

File: src/Stats.CollectAzureCdnLogs/Job.cs [hunk 2]
```diff
@@ -239,18 +239,18 @@ private static Uri ValidateFtpUri(string serverUrl)
...
                         // Trace, but ignore the failing file. Other files should go through just fine.
-                        _logger.LogWarning("Invalid raw log filename.", exception);
+                        _logger.LogWarning("Invalid raw log filename. {Exception}", exception);
                     }
                 }
             }
             catch (Exception exception)
             {
-                _logger.LogCritical("Job run failed!", exception);
+                _logger.LogCritical("Job run failed {Exception}!", exception);
```
Comment (scottbommarito) [inline src/Stats.CollectAzureCdnLogs/Job.cs]: `Job failed to initialize! {Exception}` is probably what you want here.

**Comment by scottbommarito** [comment]
A couple nits, LGTM!

**Comment by scottbommarito** [comment]
Also, one side comment, I thought it might be nice if in the `JobRunner.JobLoop` there was a message right before the job is started so we can tell if a freeze happened inside the job or if the job loop never restarted.

**Comment by skofman1** [comment]
There is a trace: Trace.WriteLine("Job run started...");

**Comment by scottbommarito** [comment]
Yeah there's a trace, but it seems like `Trace.WriteLine` doesn't get outputted to AI. As far as I can tell, only `Trace.TraceInformation`, `Trace.TraceWarning`, and `Trace.TraceError` messages show up as CustomEvents in AI.

**Comment by shishirx34** [comment]
Looks Good.

**Comment by maartenba** [comment]
Looks good to me!

**Comment by skofman1** [comment]
Merged

**Pull request closed without merging**